### ISSUE TITLE
Brave News: Support channel subscriptions and grouping subscriptions by region

### ISSUE THREAD

**Issue by petemill** [brave/brave-browser#24323]
As part of a range of new features for Brave News, the format for serializing brave news source preferences needs to support more features. The content in `sources.json` will contain a new `channels` attribute which will contain a 1 or more channels (a.k.a categories) each source appears in.

## The serialization format should:
- Remember which "channels" a user is subscribed to
- Continue remembering which "Publisher" the user has subscribed to, or unsubscribed from, individually. (Unsubscriptions will hide a Publisher even if they are part of a subscribed-to channel).
- Groups by region all subscription channels and sources, except direct feeds (a.k.a RSS feeds)

## Migration:
- Automatically subscribe new and existing users to the `"Top Sources"` channel. This channel behaves like the existing feature that `enabled: boolean` provides as part of a Publisher. There is now no concept of a singular _default enabled_ for a Publisher. In the future, a new onboarding UX will allow the user to choose custom channels and Publishers initially and never be subscribed to the "Top Sources" channel.
- Supports existing `sources.json` format as well as new `channels` attribute
  - When parsing items in `source.json`, If item doesn't have the `channels` attribute, but does have the `enabled` attribute, then assign the item the `"Top Sources"` channel to the item and add the `category` value as a channel instead.

## Current pref structure:
Simply stores whether a Publisher has been specifically opted-out or specifically opted-in by the user
```
'brave.today.sources': {
      [publisherId]: boolean
}
```

## Proposed pref structure:
```
'brave.today.subscriptions': {
     ...
}
```

**Comment by petemill** [comment]
Let's find out from the team whether a publisher ID (a guid) can be consistent for a publisher if it appears in multiple regions. @fallaciousreasoning bought up the good point of keeping an unusbscription to a channel when the user switches between regions.

**Comment by fallaciousreasoning** [comment]
@petemill I've started on the categories stuff (at last :D). I know the original plan was to introduce a `Top Sources` category and add all the enabled by default feeds to it. However, I noticed there's already a `Top News` category, and nearly 3/4 of it is subscribed by default (28/40 sources). Unfortunately, there are a further 59 sources which are not in `Top News` but are subscribed by default.

I worry a bit that having `Top News` and `Top Sources` will be confusing.

Maybe we should call it `Default Sources` instead? Either way, I think this default category should exist on the server, and that way we can configure the defaults over time.
